File: notification-adapter-in/src/main/java/com/example/notification/adapter/in/rest/dto/SendNotificationResponse.java
package com.example.notification.adapter.in.rest.dto;

import com.example.notification.application.dto.SendNotificationResult;
import com.example.notification.domain.channel.ChannelType;
import com.example.notification.domain.notification.NotificationStatus;
import java.util.List;
import java.util.UUID;

public record SendNotificationResponse(
        UUID notificationId,
        NotificationStatus status,
        List<ChannelType> dispatchedChannels,
        String suppressionReason) {

    public static SendNotificationResponse from(SendNotificationResult result) {
        return new SendNotificationResponse(
                result.notificationId(),
                result.status(),
                result.dispatchedChannels(),
                result.suppressionReason());
    }
}

File: notification-adapter-in/src/main/kotlin/com/example/notification/adapter/in/rest/dto/SendNotificationResponse.kt
package com.example.notification.adapter.`in`.rest.dto

import com.example.notification.application.dto.SendNotificationResult
import com.example.notification.domain.channel.ChannelType
import com.example.notification.domain.notification.NotificationStatus
import java.util.UUID

@JvmRecord
data class SendNotificationResponse(
    val notificationId: UUID,
    val status: NotificationStatus,
    val dispatchedChannels: List<ChannelType>,
    val suppressionReason: String?,
) {

    companion object {
        @JvmStatic
        fun from(result: SendNotificationResult): SendNotificationResponse =
            SendNotificationResponse(
                result.notificationId,
                result.status,
                result.dispatchedChannels,
                result.suppressionReason,
            )
    }
}
